File: android/core/model/src/main/java/com/bolsaterminal/core/model/Bot.kt
package com.bolsaterminal.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class BotMode {
    @SerialName("conservative") Conservative,
    @SerialName("moderate") Moderate,
    @SerialName("aggressive") Aggressive,
}

@Serializable
data class BotConfig(
    val enabled: Boolean,
    val mode: BotMode,
    val targetSymbols: List<String>,
    val scanInterval: Double,
)

@Serializable
data class BotParams(
    val minScore: Double,
    val maxRSI: Double,
    val stopLossPct: Double,
    val takeProfitPct: Double,
    val maxPositions: Int,
    val positionSizePct: Double,
    val sellScore: Double,
)

@Serializable
data class BotStatus(
    val config: BotConfig,
    val params: BotParams,
    val scanCount: Int,
    val isRunning: Boolean,
    val logCount: Int,
)

@Serializable
enum class BotLogAction {
    @SerialName("BUY") Buy,
    @SerialName("SELL") Sell,
    @SerialName("HOLD") Hold,
    @SerialName("SCAN") Scan,
    @SerialName("INFO") Info,
    @SerialName("ERROR") Error,
}

@Serializable
data class BotLogEntry(
    val id: String,
    val timestamp: Double,
    val action: BotLogAction,
    val symbol: String? = null,
    val price: Double? = null,
    val quantity: Double? = null,
    val reason: String,
    val score: Double? = null,
    val rsi: Double? = null,
    val pnl: Double? = null,
)

File: macos/BolsaTerminal/Core/Models/Bot.swift
import Foundation

enum BotMode: String, Codable, CaseIterable {
    case conservative
    case moderate
    case aggressive

    var label: String {
        switch self {
        case .conservative: return "Conservador"
        case .moderate: return "Moderado"
        case .aggressive: return "Agresivo"
        }
    }
}

struct BotConfig: Codable, Hashable {
    let enabled: Bool
    let mode: BotMode
    let targetSymbols: [String]
    let scanInterval: Double
}

struct BotParams: Codable, Hashable {
    let minScore: Double
    let maxRSI: Double
    let stopLossPct: Double
    let takeProfitPct: Double
    let maxPositions: Int
    let positionSizePct: Double
    let sellScore: Double
}

struct BotStatus: Codable, Hashable {
    let config: BotConfig
    let params: BotParams
    let scanCount: Int
    let isRunning: Bool
    let logCount: Int
}

enum BotLogAction: String, Codable {
    case buy = "BUY"
    case sell = "SELL"
    case hold = "HOLD"
    case scan = "SCAN"
    case info = "INFO"
    case error = "ERROR"
}

struct BotLogEntry: Codable, Hashable, Identifiable {
    let id: String
    let timestamp: Double
    let action: BotLogAction
    let symbol: String?
    let price: Double?
    let quantity: Double?
    let reason: String
    let score: Double?
    let rsi: Double?
    let pnl: Double?
}
